### PR TITLE
Dotnet section

### LINKED ISSUE #71: Implement a prompt section for .NET SDK version (.NET)
## Feature Request

**Is your feature request related to a problem? Please describe.**
This feature is intended to achieve feature parity with the following Spaceship section file: https://github.com/denysdovhan/spaceship-prompt/blob/master/sections/dotnet.zsh

**Describe the solution you'd like**
Porting over the existing zsh implementation of the .NET section file, showing the current .NET version when the prompt is in the same directory as a `*.csproj`, `*.xproj`, `*.sln`, `global.json` or `project.json` file.

### PR DESCRIPTION
**Pull request by ladysamantha**
Add a .NET SDK section to the prompt.

## Description

Add a .NET SDK section based on the dotnet-cli. 

## Motivation and Context

Adds a new section to have greater parity with the ZSH version. 

Closes #71 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots (if appropriate):
screen shot 2018-10-12 at 19 18 59

## How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**

## Checklist:

- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by ladysamantha** [comment]
More tests need to be added for this section, but I wanted to open the PR anyway just to get it started!

**Comment by ladysamantha** [comment]
Errrm I also just noticed that this PR added my other commits for some reason. If you need me to close this and make a new one, just let me know!

**Comment by matchai** [comment]
Looks good so far! 👍 
If you put "WIP", "Work In Progress" or "Do Not Merge" in the title of the PR, our [WIP GitHub App](https://github.com/apps/wip) will stop me from accidentally merging it. Just a heads up. 😄 

No need to worry about the commit history. I'll be squash-merging it. 🥞

**Comment by matchai** [comment]
Hey @ladysamantha! 👋 
It's been a little over a month. Just want to check up with you and see if you're still working on this.

**Comment by ladysamantha** [comment]
Hello 👋 I'm sorry for the delay, I was working on it and then got sidetracked by work and the holiday and haven't had a lot of time recently to work on it.

I have time this week to add the tests. I've blocked out time on my calendar to finish it.

**Comment by matchai** [comment]
It happens to everyone. No worries. 😄 
Looking forward to reviewing your PR once you're done with it. 👍

File: tests/__sf_section_dotnet.test.fish [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	cd /tmp/tmp-spacefish
+end
+
+function teardown
+	rm -rf /tmp/tmp-spacefish
+end
+
+test "Prints section"
+	(
+		touch
```
Comment (Snuggle) [inline tests/__sf_section_dotnet.test.fish]: This touch is causing a silent error in the tests, just as a heads-up! :slightly_smiling_face: 
image

**Comment by matchai** [comment]
Heya @ladysamantha!
Just checking in to see if you're still interested in finishing this PR up. 😄
If you're too busy, I wouldn't mind finishing it up. 

Whatever works for you 👍

**Comment by ladysamantha** [comment]
Sorry again for the delay; should be good to continue working on this PR during the holiday break.
Just pushed the tests. Let me know what I need to fix 😄

**Comment by matchai** [comment]
Glad to hear it! 😄 
It looks like you need to address a merge conflict in `fish_prompt.fish` before CI will run.

File: docs/Options.md [hunk 2]
```diff
@@ -290,6 +290,18 @@ Rust section is shown only in directories that contain `Cargo.toml` or any other
 | `SPACEFISH_RUST_COLOR` | `red` | Color of Rust section |
 | `SPACEFISH_RUST_VERBOSE_VERSION` | `false` | Show what branch is being used, if any. (Beta, Nightly) |
 
+### .NET (`dotnet`)
+
+.NET section is shown only in directories that contains a `project.json` or `global.json` file, or a file with one of these extensions: `.csproj`, `.xproj` or `.sln`.
+
+| Variable | Default | Meaning |
+| :------- | :-----: | ------- |
+| `SPACESHIP_DOTNET_SHOW` | `true` | Current .NET section |
```
Comment (matchai) [inline docs/Options.md]: These variables should be switched to their "SPACEFISH" equivalent. 🚀🐟

File: tests/__sf_section_dotnet.test.fish [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		rm -f /tmp/tmp-spacefish/global.json
+		rm -f '/tmp/tmp-spacefish/*.sln'
+		rm -f '/tmp/tmp-spacefish/*.csproj'
+		rm -f '/tmp/tmp-spacefish/*.xproj'
+	) = (__sf_section_dotnet)
+end
+
+test "Prints section if project.json is present"
+	(
+		touch /tmp/tmp-spacefish/project.json
```
Comment (matchai) [inline tests/__sf_section_dotnet.test.fish:27]: nit: I would add a newline separating the setup from the test output like in other tests in this file 😀

**Review by matchai — APPROVED** [review]
The PR looks great! Just a couple small comments and we'll merge it in. 😁

File: docs/Options.md (hunk 2 above)
Comment (ladysamantha) [reply docs/Options.md]: Ohhh good catch. Fixed

File: tests/__sf_section_dotnet.test.fish (hunk 3 above)
Comment (ladysamantha) [reply tests/__sf_section_dotnet.test.fish:27]: Done.

File: tests/__sf_section_dotnet.test.fish (hunk 1 above)
Comment (ladysamantha) [reply tests/__sf_section_dotnet.test.fish]: Should be fixed now

**Pull request merged**